DO_NOT_SUBMIT: coroutine-related test code to attempt to identify why errors are being suppressed silently.

jonmolle opened the pull request.

jonmolle committed:
update common-jvm version to include a passable context

jonmolle committed:
use a limited view on Dispatchers.IO for encryption

SanjayVas reviewed (changes requested):
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431)* status: all files reviewed, 2 unresolved discussions (waiting on @jonmolle)

___
*[`build/repositories.bzl` line 27 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2HFs-UFrw6frpNj5:-NoT2HFs-UFrw6frpNj6:b-t2d93s) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/build/repositories.bzl#L27)):*
> ```Starlark
>     """Imports all direct dependencies for wfa_measurement_system."""
> 
> # TODO(jmolle): Update this to a release that supports passing
> ```

1. wfa_repo_archive supports `commit` instead of `version`.
2. The common pattern is to add a DO_NOT_SUBMIT comment with TODO format referencing the PR. e.g.`DO_NOT_SUBMIT(world-federation-of-advertisers/common-jvm#XXX): Use version.`
3. I thought @stevenwarejones said he abandoned that branch?
___
*[`src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt` line 124 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2dbI-f2zML1wViTL:-NoT2dbI-f2zML1wViTM:b-d67eev) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt#L124)):*
> ```Kotlin
>       // Is the underlying code thread-safe? Set to 1 for testing, the idea being this should be
>       // a special view just for encrypt/decrypt.
>       TinkKeyStorageProvider(Dispatchers.IO.limitedParallelism(1), s3Bucket))
> ```

Is this trying to debug not getting an exception? I'm much more suspicious about the calling code's exception handling behavior w.r.t. coroutine exception propagation than I am with the thread safety of Tink Aead impls.


<!-- Sent from Reviewable.io -->

jonmolle committed:
Adds a catch inside the supervisor scope to handle uncaught exceptions

- attempting to check to see if withContext() is propagating exceptions
  up to the parent, which is a supervisor scope, which suppresses it
  before the error handling in ExchangeStepLauncher can catch it.

jonmolle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @SanjayVas and @stevenwarejones)

___
*[`src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt` line 124 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2dbI-f2zML1wViTL:-NoTCsqABUGt-Ujkxcow:bovzhui) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt#L124)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Is this trying to debug not getting an exception? I'm much more suspicious about the calling code's exception handling behavior w.r.t. coroutine exception propagation than I am with the thread safety of Tink Aead impls.
</blockquote></details>

I've been looking at this, but the code should throw a GeneralSecurityException (wrapping whatever it gets from AWS), which should be propagated back up via the withContext call. The resulting error should be logged/consumed by invalidateAttempt() or, at worst, logAndSuppressExceptionSuspend. The only thing I can think of is that somehow withContext is propagated up through to the launch statement that is under the supervisor scope, so the error is suppressed there with no handling. I've duplicated some error handling there to check.
___
*[`build/repositories.bzl` line 27 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2HFs-UFrw6frpNj5:-NoTCg9A5-l9oaRsckJn:b-b297pi) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/build/repositories.bzl#L27)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

1. wfa_repo_archive supports `commit` instead of `version`.
2. The common pattern is to add a DO_NOT_SUBMIT comment with TODO format referencing the PR. e.g.`DO_NOT_SUBMIT(world-federation-of-advertisers/common-jvm#XXX): Use version.`
3. I thought @stevenwarejones said he abandoned that branch?
</blockquote></details>

This is just a proposed solution to a problem that seems to have been solved based on another discussion. I'll add DO_NOT_SUMBIT


<!-- Sent from Reviewable.io -->

jonmolle reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @SanjayVas and @stevenwarejones)

___
*[`src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt` line 124 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2dbI-f2zML1wViTL:-NoTfcjw49QxLF20U5FN:b-2o68) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt#L124)):*
<details><summary><i>Previously, jonmolle wrote…</i></summary><blockquote>

I've been looking at this, but the code should throw a GeneralSecurityException (wrapping whatever it gets from AWS), which should be propagated back up via the withContext call. The resulting error should be logged/consumed by invalidateAttempt() or, at worst, logAndSuppressExceptionSuspend. The only thing I can think of is that somehow withContext is propagated up through to the launch statement that is under the supervisor scope, so the error is suppressed there with no handling. I've duplicated some error handling there to check.
</blockquote></details>

As a followup here, this should be caught by the catch in ExchangeTaskExecutor.execute, which should properly mark a failure, but clearly that's being skipped, as we'd see it in the logs.


<!-- Sent from Reviewable.io -->

stevenwarejones reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @SanjayVas)

___
*[`src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt` line 124 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/1431#-NoT2dbI-f2zML1wViTL:-NoXpm8g0U8dK6a4BAZn:b-1cnrj1) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/d0580d47aabd1b46a4c5c574817c8a6de6268ea2/src/main/kotlin/org/wfanet/panelmatch/client/deploy/example/aws/AwsExampleDaemon.kt#L124)):*
<details><summary><i>Previously, jonmolle wrote…</i></summary><blockquote>

As a followup here, this should be caught by the catch in ExchangeTaskExecutor.execute, which should properly mark a failure, but clearly that's being skipped, as we'd see it in the logs.
</blockquote></details>

are you able to write a test that replicates this? It was unclear to me that what you're describing is what is definitely happening given the logs we saw.


<!-- Sent from Reviewable.io -->
